File: code/richediting/src/main/java/org/fbme/ide/richediting/adapters/common/actions/RelayoutAction.java
package org.fbme.ide.richediting.adapters.common.actions;

import jetbrains.mps.openapi.editor.cells.EditorCell;
import org.fbme.ide.richediting.adapters.ecc.actions.RelayoutECCAction;
import org.fbme.ide.richediting.adapters.fbnetwork.actions.RelayoutFBNetworkAction;

public class RelayoutAction implements Action {
    private final EditorCell cell;

    public RelayoutAction(EditorCell cell) {
        this.cell = cell;
    }

    @Override
    public void apply() {
        String typeName = cell.getSNode().getConcept().getName();
        if (typeName.equals("BasicFBTypeDeclaration")) {
            new RelayoutECCAction(cell).apply();
        } else if (typeName.equals("CompositeFBTypeDeclaration")) {
            new RelayoutFBNetworkAction(cell).apply();
        }
    }
}

File: code/richediting/src/main/kotlin/org/fbme/ide/richediting/adapters/common/actions/RelayoutAction.kt
package org.fbme.ide.richediting.adapters.common.actions

import jetbrains.mps.openapi.editor.cells.EditorCell
import org.fbme.ide.richediting.adapters.ecc.actions.RelayoutECCAction
import org.fbme.ide.richediting.adapters.fbnetwork.actions.RelayoutFBNetworkAction

class RelayoutAction(private val cell: EditorCell) : Action {
    override fun apply() {
        val typeName = cell.sNode.concept.name
        if (typeName == "BasicFBTypeDeclaration") {
            RelayoutECCAction(cell).apply()
        } else if (typeName in listOf(
                "CompositeFBTypeDeclaration",
                "ApplicationDeclaration",
                "SubapplicationTypeDeclaration",
                "ResourceDeclaration"
            )
        ) {
            RelayoutFBNetworkAction(cell).apply()
        }
    }
}
